[v2.0.0-rc.0] Broken support for extended fence language

This is a regression of the feature introduced by a5d0ccec6f14f4808a535c0f06273d1e9892be15. 

For example, using the same input as the unit test:

````
```js more words are ignored
var answer = 6 * 7;
```

```js
var answer = 6 * 7;
```
````

Running ESLint on the markdown above:

```
> eslint *.md


test.md
  6:1  error  Unexpected var, use let or const instead  no-var

✖ 1 problem (1 error, 0 warnings)
```

Should report 2 errors with the following config:

```js
module.exports = {
  extends: 'plugin:markdown/recommended',
  rules: {
    'no-var': 'error'
  }
};
```

I haven't been able to run tests locally (failed to `npm install` this repo on Windows) but I think that the [associated unit test](https://github.com/eslint/eslint-plugin-markdown/blob/d30c50f46237af2fdef0a8a21fb547ed8e6c4d80/tests/lib/processor.js#L271-L280) doesn't guarantee that the returned block will be processed by ESLint. I guess the returned `block.filename` is `'0.js more words are ignored'` which doesn't pass the glob matching on this virtual block filename.

--- a/lib/processor.js
+++ b/lib/processor.js
@@ -252,7 +252,9 @@ function preprocess(text) {
     });
 
     return blocks.map((block, index) => ({
-        filename: `${index}.${block.lang.trimStart().split(" ")[0]}`,
+
+        // TODO: Use trimStart() after dropping Node.js 8.x support.
+        filename: `${index}.${block.lang.replace(/^\s+/u, "").split(" ")[0]}`,
         text: [
             ...block.comments,
             block.value,

--- a/lib/processor.js
+++ b/lib/processor.js
@@ -252,7 +252,7 @@ function preprocess(text) {
     });
 
     return blocks.map((block, index) => ({
-        filename: `${index}.${block.lang}`,
+        filename: `${index}.${block.lang.trimStart().split(" ")[0]}`,
         text: [
             ...block.comments,
             block.value,

--- a/lib/processor.js
+++ b/lib/processor.js
@@ -252,9 +252,7 @@ function preprocess(text) {
     });
 
     return blocks.map((block, index) => ({
-
-        // TODO: Use trimStart() after dropping Node.js 8.x support.
-        filename: `${index}.${block.lang.replace(/^\s+/u, "").split(" ")[0]}`,
+        filename: `${index}.${block.lang.trim().split(" ")[0]}`,
         text: [
             ...block.comments,
             block.value,

--- a/tests/lib/processor.js
+++ b/tests/lib/processor.js
@@ -297,6 +297,18 @@ describe("processor", () => {
             assert.strictEqual(blocks[0].filename, "0.js");
         });
 
+        it("should ignore trailing whitespace in the info string", () => {
+            const code = [
+                "```js    ",
+                "var answer = 6 * 7;",
+                "```"
+            ].join("\n");
+            const blocks = processor.preprocess(code);
+
+            assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.js");
+        });
+
         it("should find code fences not surrounded by blank lines", () => {
             const code = [
                 "<!-- eslint-disable -->",

--- a/tests/lib/processor.js
+++ b/tests/lib/processor.js
@@ -222,6 +222,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.js");
         });
 
         it("should find code fences with javascript info string", () => {
@@ -233,6 +234,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.javascript");
         });
 
         it("should find code fences with node info string", () => {
@@ -244,6 +246,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.node");
         });
 
         it("should find code fences with jsx info string", () => {
@@ -255,6 +258,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.jsx");
         });
 
         it("should find code fences ignoring info string case", () => {
@@ -266,6 +270,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.JavaScript");
         });
 
         it("should ignore anything after the first word of the info string", () => {
@@ -277,6 +282,19 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.js");
+        });
+
+        it("should ignore leading whitespace in the info string", () => {
+            const code = [
+                "```  js ignores leading whitespace",
+                "var answer = 6 * 7;",
+                "```"
+            ].join("\n");
+            const blocks = processor.preprocess(code);
+
+            assert.strictEqual(blocks.length, 1);
+            assert.strictEqual(blocks[0].filename, "0.js");
         });
 
         it("should find code fences not surrounded by blank lines", () => {
@@ -293,6 +311,8 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks.length, 2);
+            assert.strictEqual(blocks[0].filename, "0.js");
+            assert.strictEqual(blocks[1].filename, "1.js");
         });
 
         it("should return the source code in the block", () => {

